pkg/genericcli/printers/csv.go

```diff
@@ -0,0 +1,55 @@
+package printers
+
+import (
+	"encoding/csv"
+	"io"
+
+	"github.com/jszwec/csvutil"
+)
+
+const defaultDelimiter = ';'
+
+type CSVPrinter struct {
+	c   *CSVPrinterConfig
+	out io.Writer
+}
+
+type CSVPrinterConfig struct {
+	// AutoHeader will generate headers during print, default is go standard ("false")
+	AutoHeader bool
+	// Delimiter the char to separate the columns, default is ";"
+	Delimiter rune
+}
+
+func NewCSVPrinter(config *CSVPrinterConfig) *CSVPrinter {
+	if config.Delimiter == 0 {
+		config.Delimiter = defaultDelimiter
+	}
+
+	return &CSVPrinter{
+		c: config,
+	}
+}
+
+func (cp *CSVPrinter) WithOut(out io.Writer) *CSVPrinter {
```

Comment:
Why do you have a Option for setting the Writer, but a Config for header and delimiter, seems a two fold approach. I think either config or options for all configurable settings.

Reply:
edit out as an output option as well